File: package/ios/KetchAdIdentifiers.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(KetchAdIdentifiers, NSObject)

RCT_EXTERN_METHOD(getIdfv:(RCTPromiseResolveBlock)resolve
                  reject:(RCTPromiseRejectBlock)reject)

@end

File: package/ios/KetchAdIdentifiers.swift
import Foundation
import React
import UIKit

@objc(KetchAdIdentifiers)
class KetchAdIdentifiers: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { false }

  @objc func getIdfv(
    _ resolve: @escaping RCTPromiseResolveBlock,
    reject: @escaping RCTPromiseRejectBlock
  ) {
    resolve(UIDevice.current.identifierForVendor?.uuidString)
  }
}
